src/test/java/com/aws/greengrass/clientdevices/auth/CertificateManagerTest.java

```diff
@@ -69,25 +87,88 @@ public class CertificateManagerTest {
-                mockCertExpiryMonitor, mockShadowMonitor, Clock.systemUTC(), clientFactory);
+                mockCertExpiryMonitor, mockShadowMonitor, Clock.systemUTC(), clientFactoryMock, securityServiceMock);
 
         CertificatesConfig certificatesConfig = new CertificatesConfig(
-                Topics.of(new Context(), KernelConfigResolver.CONFIGURATION_CONFIG_KEY, null));
+                Topics.of(new Context(), CONFIGURATION_CONFIG_KEY, null));
         CAConfiguration caConfiguration = CAConfiguration.from(
-                Topics.of(new Context(), KernelConfigResolver.CONFIGURATION_CONFIG_KEY, null));
+                Topics.of(new Context(), CONFIGURATION_CONFIG_KEY, null));
 
         certificateManager.updateCertificatesConfiguration(certificatesConfig);
         certificateManager.updateCAConfiguration(caConfiguration);
         certificateManager.update("");
     }
 
+    @AfterEach
+    void afterEach() {
+        reset(securityServiceMock);
+    }
+
+    @Test
+    void GIVEN_customCAConfiguration_WHEN_configureCustomCA_THEN_returnsCustomCA() throws Exception {
+        // Given
+        Instant now = Instant.now();
+        KeyPair keyPair = CertificateStore.newRSAKeyPair(2048);
+        X509Certificate caCertificate = CertificateHelper.createCACertificate(
+                keyPair, Date.from(now), Date.from(now.plusSeconds(10)), "Custom Core CA");
+
+        URI privateKeyUri = new URI("file:///private.key");
+        URI certificateUri = new URI("file:///certificate.pem");
+
+        Topics configuration = Topics.of(new Context(), CONFIGURATION_CONFIG_KEY, null);
+        configuration.lookup(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC, CA_PRIVATE_KEY_URI)
+                .withValue(privateKeyUri.toString());
+        configuration.lookup(CONFIGURATION_CONFIG_KEY, CERTIFICATE_AUTHORITY_TOPIC, CA_CERTIFICATE_URI)
+                .withValue(certificateUri.toString());
+
+        CAConfiguration caConfiguration = CAConfiguration.from(configuration);
+        certificateManager.updateCAConfiguration(caConfiguration);
+
+        // TODO: Write the actual certificate to the file system and avoid mocking the security service. Doing
+        //  this is a bad given we are exposing implementation details on the test.
+        when(securityServiceMock.getKeyPair(privateKeyUri, certificateUri)).thenReturn(keyPair);
+        when(securityServiceMock.getCertificateChain(privateKeyUri, certificateUri))
+                .thenReturn(new X509Certificate[]{caCertificate});
+
+        // When
+        certificateManager.configureCustomCA();
+
+        // Then
+        List<String> caPemStrings = certificateManager.getCACertificates();
+        String caPem = caPemStrings.get(0);
+        assertEquals(caPem, CertificateHelper.toPem(caCertificate));
+    }
+
+    @Test
+    void Given_defaultCAConfiguration_THEN_returnsAutoGeneratedCA() throws Exception {
```

Comment:
Just ensures that when the old config produces a default ca and not the one we are providing